ngraph/core/src/op/range.cpp

```diff
@@ -26,6 +26,230 @@
+        break;
+    case element::Type_t::f64:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::f64>());
+        break;
+    case element::Type_t::i8:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i8>());
+        break;
+    case element::Type_t::i16:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i16>());
+        break;
+    case element::Type_t::i32:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i32>());
+        break;
+    case element::Type_t::i64:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::i64>());
+        break;
+    case element::Type_t::u8:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u8>());
+        break;
+    case element::Type_t::u16:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u16>());
+        break;
+    case element::Type_t::u32:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u32>());
+        break;
+    case element::Type_t::u64:
+        *val = static_cast<T>(*tensor->get_data_ptr<element::Type_t::u64>());
+        break;
+    default: return false;
+    }
+    return true;
+}
+
+template <element::Type_t ET>
+bool evaluate_v4_range(const HostTensorPtr& out,
+                       const HostTensorPtr& start,
+                       const HostTensorPtr& stop,
+                       const HostTensorPtr& step)
+{
+    using T = typename element_type_traits<ET>::value_type;
+    T start_val;
+    T stop_val;
+    T step_val;
+    if (!(get_casted_value<T>(start, &start_val) && get_casted_value<T>(stop, &stop_val) &&
+          get_casted_value<T>(step, &step_val)))
+    {
+        return false;
+    }
+
+    int64_t out_size = 0;
+
+    int64_t steps = static_cast<int64_t>(std::ceil(double(stop_val - start_val) / step_val));
+    if (steps > 0)
+    {
+        out_size = steps;
+    }
+    Shape out_shape = Shape({static_cast<size_t>(out_size)});
+    out->set_shape(out_shape);
+    runtime::reference::range(&start_val, &step_val, out_shape, out->get_data_ptr<ET>());
+    return true;
+}
+
+bool op::v4::Range::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const
+{
+    HostTensorPtr out = outputs[0];
+    HostTensorPtr start = inputs[0];
+    HostTensorPtr stop = inputs[1];
+    HostTensorPtr step = inputs[2];
+    switch (m_output_type)
+    {
+    case element::Type_t::bf16:
+        return evaluate_v4_range<element::Type_t::bf16>(out, start, stop, step);
+        break;
+    case element::Type_t::f16:
+        return evaluate_v4_range<element::Type_t::f16>(out, start, stop, step);
+        break;
+    case element::Type_t::f32:
+        return evaluate_v4_range<element::Type_t::f32>(out, start, stop, step);
+        break;
+    case element::Type_t::f64:
+        return evaluate_v4_range<element::Type_t::f64>(out, start, stop, step);
```

Comment:
Do we need all this data types?

For example IE doesn't support f64...

Could you measure the impact on binary size from this operation?

Reply:
@ilyachur I measured size difference, it is 20 480 bytes.